src/pythie_serving/table_wrapper.py

```diff
@@ -1,92 +1,72 @@
 import csv
 import json
-import logging
 from pathlib import Path
-from typing import Any, Dict
 
-import grpc
 import numpy as np
+from numpy.typing import NDArray
 
-from .exceptions import PythieServingException
-from .tensorflow_proto.tensorflow_serving.apis import (
-    predict_pb2,
-    prediction_service_pb2_grpc,
+from pythie_serving.abstract_wrapper import (
+    AbstractPythieServingPredictionServiceServicer,
+    ModelSpecs,
 )
-from .tensorflow_proto.tensorflow_serving.config import model_server_config_pb2
-from .utils import get_csv_type, parse_sample
 
+from .exceptions import PythieServingException
+from .tensorflow_proto.tensorflow_serving.config.model_server_config_pb2 import (
+    ModelConfig,
+)
+from .utils import get_csv_type
 
-class TablePredictionServiceServicer(prediction_service_pb2_grpc.PredictionServiceServicer):
-    def __init__(
-        self,
-        *,
-        logger: logging.Logger,
-        model_server_config: model_server_config_pb2.ModelServerConfig,
-    ):
-        self.logger = logger
-        self.table_map: Dict[str, Any] = {}
-        for table_config in model_server_config.model_config_list.config:
 
-            table_path = Path().joinpath(table_config.base_path, table_config.name + ".csv")
-            if not table_path.exists():
-                raise PythieServingException(f"CSV table {table_config.name} not found " f"at {table_path}")
+class TablePredictionServiceServicer(AbstractPythieServingPredictionServiceServicer):
+    def _create_model_specs(self, model_config: ModelConfig) -> ModelSpecs:
 
-            metadata_path = Path().joinpath(table_config.base_path, "metadata.json")
-            if not metadata_path.exists():
-                raise PythieServingException(f"Metadata file metadata.json not found " f"at {metadata_path}")
+        table_path = Path().joinpath(model_config.base_path, model_config.name + ".csv")
+        if not table_path.exists():
+            raise PythieServingException(f"CSV table {model_config.name} not found " f"at {table_path}")
 
-            with open(metadata_path) as f:
-                metadata = json.load(f)
+        metadata_path = Path().joinpath(model_config.base_path, "metadata.json")
+        if not metadata_path.exists():
+            raise PythieServingException(f"Metadata file metadata.json not found " f"at {metadata_path}")
```

Comment:
This might not be useful, as this file is copy-paste'd at model-download (pod initialization), but maybe we still want to keep it, as pythie-serving can be used without the model-download step.
If we want to keep it, we should add this to all wrapper.

Reply:
same, I'll move it to the abstract wrapper